src/main/java/org/spdx/maven/SpdxDependencyInformation.java

```diff
@@ -225,14 +231,17 @@ else if ( scope.equals( "test" ) )
     }
 
     /**
-     * Create an SPDX Document from a POM file stored in the Maven repository
-     *
+     * Create an SPDX Document using the mavenProjectBuilder to resolve properties
+     * including inherited properties
      * @param artifact Maven dependency artifact
-     * @return
+     * @param mavenProjectBuilder project builder for the repo containing the POM file
+     * @param session Maven session for building the project
+     * @return SPDX Package build from the MavenProject metadata
      * @throws LicenseMapperException
      * @throws InvalidSPDXAnalysisException 
      */
-    private SpdxElement createSpdxPackage( Artifact artifact ) throws LicenseMapperException, InvalidSPDXAnalysisException
+    private SpdxElement createSpdxPackage( Artifact artifact, 
+                                           ProjectBuilder mavenProjectBuilder, MavenSession session ) throws LicenseMapperException, InvalidSPDXAnalysisException
     {
         log.debug( "Creating SPDX package for artifact " + artifact.getArtifactId() );
         if ( artifact.getFile() == null )
```

Comment:
I'm assuming that we can get the file directly from the artifact and not run it through the Maven project builder.  @alerosmile - let me know if you agree

Reply:
@goneall - sorry I don't get what you mean by 'getting the file from the artifact'.

Do you mean that the following code from the SO answer is not required? `Artifact pomArtifact = repositorySystem.createProjectArtifact(groupId, artifactId, version);` If yes then I agree.

I've reviewed and tested the PR. Looks good to me. Properties are now resolved properly. Thanks!

Reply:
> sorry I don't get what you mean by 'getting the file from the artifact'.

The `artifact` used to get the file is obtained from the call to `mavenProject.getArtifacts()` where `mavenProject` is the project we're producing the SPDX document for.  The `artifact` is passed in as a parameter to `ProjectBuilder` to create the project we use to build the SPDX package.  I'm assuming the call to `artifact.getFile()` will return the same file as the call to `mavenProjectBuilder.build(artifact, session.getProjectBuildingRequest()).getProject().getArtifact().getFile()` in which case the code should work.  The current implementation delays building the project until it is really needed.

> I've reviewed and tested the PR. Looks good to me. Properties are now resolved properly. Thanks!

Thanks @alerosmile for testing it out.

Reply:
The file obtained from the call to `mavenProjectBuilder.build(artifact, session.getProjectBuildingRequest()).getProject().getArtifact().getFile()` is `null`. That actually makes sense because the model is usually used during the build of that file. The model is used to create multiple artifacts from the same POM (e.g. `.jar`, `-sources.jar`). The artifact obtained from the call to `mavenProject.getArtifacts()` is just one of these, the one defined in the pom of the project to be built.

You need the model used to build the artifact. You don't need that file but if you do, you have to obtain it directly from the artifact.

Reply:
> The current implementation delays building the project until it is really needed.

@goneall - what do you mean by this? The projects of the dependencies?

Reply:
@alerosmile Yes - the projects of the dependencies.